### ISSUE TITLE
Upgrade postcss package to v8.2.10

### ISSUE THREAD

**Issue by kyoto** [thanos-io/thanos#4065]
postcss before v8.2.10 (used by the React UI) is affected by CVE-2021-23368 (see https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-23368)

Output from `npm list postcss`:

```
+-- react-scripts@3.4.4
| +-- css-loader@3.4.2
| | +-- icss-utils@4.1.1
| | | `-- postcss@7.0.35  deduped
| | +-- postcss@7.0.35
| | +-- postcss-modules-extract-imports@2.0.0
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-modules-local-by-default@3.0.3
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-modules-scope@2.2.0
| | | `-- postcss@7.0.35  deduped
| | `-- postcss-modules-values@3.0.0
| |   `-- postcss@7.0.35  deduped
| +-- optimize-css-assets-webpack-plugin@5.0.3
| | `-- cssnano@4.1.11
| |   +-- cssnano-preset-default@4.0.8
| |   | +-- css-declaration-sorter@4.0.1
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- cssnano-util-raw-cache@4.0.1
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss@7.0.35  deduped
| |   | +-- postcss-calc@7.0.5
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-colormin@4.0.3
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-convert-values@4.0.1
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-discard-comments@4.0.2
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-discard-duplicates@4.0.2
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-discard-empty@4.0.1
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-discard-overridden@4.0.1
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-merge-longhand@4.0.11
| |   | | +-- postcss@7.0.35  deduped
| |   | | `-- stylehacks@4.0.3
| |   | |   `-- postcss@7.0.35  deduped
| |   | +-- postcss-merge-rules@4.0.3
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-minify-font-values@4.0.2
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-minify-gradients@4.0.2
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-minify-params@4.0.2
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-minify-selectors@4.0.2
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-normalize-charset@4.0.1
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-normalize-display-values@4.0.2
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-normalize-positions@4.0.2
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-normalize-repeat-style@4.0.2
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-normalize-string@4.0.2
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-normalize-timing-functions@4.0.2
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-normalize-unicode@4.0.1
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-normalize-url@4.0.1
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-normalize-whitespace@4.0.2
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-ordered-values@4.1.2
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-reduce-initial@4.0.3
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-reduce-transforms@4.0.2
| |   | | `-- postcss@7.0.35  deduped
| |   | +-- postcss-svgo@4.0.3
| |   | | `-- postcss@7.0.35  deduped
| |   | `-- postcss-unique-selectors@4.0.1
| |   |   `-- postcss@7.0.35  deduped
| |   `-- postcss@7.0.35  deduped
| +-- postcss-flexbugs-fixes@4.1.0
| | `-- postcss@7.0.35  deduped
| +-- postcss-loader@3.0.0
| | `-- postcss@7.0.35  deduped
| +-- postcss-normalize@8.0.1
| | +-- postcss@7.0.35  deduped
| | `-- postcss-browser-comments@3.0.0
| |   `-- postcss@7.0.35  deduped
| +-- postcss-preset-env@6.7.0
| | +-- autoprefixer@9.8.6
| | | `-- postcss@7.0.35  deduped
| | +-- css-blank-pseudo@0.1.4
| | | `-- postcss@7.0.35  deduped
| | +-- css-has-pseudo@0.10.0
| | | `-- postcss@7.0.35  deduped
| | +-- css-prefers-color-scheme@3.1.1
| | | `-- postcss@7.0.35  deduped
| | +-- postcss@7.0.35  deduped
| | +-- postcss-attribute-case-insensitive@4.0.2
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-color-functional-notation@2.0.1
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-color-gray@5.0.0
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-color-hex-alpha@5.0.3
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-color-mod-function@3.0.3
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-color-rebeccapurple@4.0.1
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-custom-media@7.0.8
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-custom-properties@8.0.11
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-custom-selectors@5.1.2
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-dir-pseudo-class@5.0.0
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-double-position-gradients@1.0.0
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-env-function@2.0.2
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-focus-visible@4.0.0
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-focus-within@3.0.0
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-font-variant@4.0.1
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-gap-properties@2.0.0
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-image-set-function@3.0.1
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-initial@3.0.2
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-lab-function@2.0.1
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-logical@3.0.0
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-media-minmax@4.0.0
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-nesting@7.0.1
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-overflow-shorthand@2.0.0
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-page-break@2.0.0
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-place@4.0.1
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-pseudo-class-any-link@6.0.0
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-replace-overflow-wrap@3.0.0
| | | `-- postcss@7.0.35  deduped
| | +-- postcss-selector-matches@4.0.0
| | | `-- postcss@7.0.35  deduped
| | `-- postcss-selector-not@4.0.1
| |   `-- postcss@7.0.35  deduped
| +-- postcss-safe-parser@4.0.1
| | `-- postcss@7.0.35  deduped
| `-- resolve-url-loader@3.1.2
|   `-- postcss@7.0.21
`-- sanitize-html@2.3.3
  `-- postcss@8.2.9
```

**Comment by simonpasquier** [comment]
A version >= v8.2.13 is required because of [CVE-2021-23382](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-23382).

**Comment by Aman-Codes** [comment]
To fix it I tried to update all the npm packages to their latest version but still found the postcss version to be less than 8.2.13 in many cases. It looks that firstly other libraries used need to update their dependencies (which might take some time) and only then we could update here.

The output of npm list postcss:

```
graph@0.1.0 /home/aman/thanos/thanos/pkg/ui/react-app
├─┬ react-scripts@4.0.3
│ ├─┬ css-loader@4.3.0
│ │ ├─┬ icss-utils@4.1.1
│ │ │ └── postcss@7.0.35  deduped
│ │ ├── postcss@7.0.35
│ │ ├─┬ postcss-modules-extract-imports@2.0.0
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-modules-local-by-default@3.0.3
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-modules-scope@2.2.0
│ │ │ └── postcss@7.0.35  deduped
│ │ └─┬ postcss-modules-values@3.0.0
│ │   └── postcss@7.0.35  deduped
│ ├─┬ optimize-css-assets-webpack-plugin@5.0.4
│ │ └─┬ cssnano@4.1.11
│ │   ├─┬ cssnano-preset-default@4.0.8
│ │   │ ├─┬ css-declaration-sorter@4.0.1
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ cssnano-util-raw-cache@4.0.1
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-calc@7.0.5
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-colormin@4.0.3
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-convert-values@4.0.1
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-discard-comments@4.0.2
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-discard-duplicates@4.0.2
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-discard-empty@4.0.1
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-discard-overridden@4.0.1
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-merge-longhand@4.0.11
│ │   │ │ ├── postcss@7.0.35  deduped
│ │   │ │ └─┬ stylehacks@4.0.3
│ │   │ │   └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-merge-rules@4.0.3
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-minify-font-values@4.0.2
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-minify-gradients@4.0.2
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-minify-params@4.0.2
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-minify-selectors@4.0.2
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-normalize-charset@4.0.1
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-normalize-display-values@4.0.2
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-normalize-positions@4.0.2
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-normalize-repeat-style@4.0.2
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-normalize-string@4.0.2
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-normalize-timing-functions@4.0.2
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-normalize-unicode@4.0.1
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-normalize-url@4.0.1
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-normalize-whitespace@4.0.2
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-ordered-values@4.1.2
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-reduce-initial@4.0.3
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-reduce-transforms@4.0.2
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ ├─┬ postcss-svgo@4.0.3
│ │   │ │ └── postcss@7.0.35  deduped
│ │   │ └─┬ postcss-unique-selectors@4.0.1
│ │   │   └── postcss@7.0.35  deduped
│ │   └── postcss@7.0.35  deduped
│ ├─┬ postcss-flexbugs-fixes@4.2.1
│ │ └── postcss@7.0.35  deduped
│ ├─┬ postcss-loader@3.0.0
│ │ └── postcss@7.0.35  deduped
│ ├─┬ postcss-normalize@8.0.1
│ │ ├── postcss@7.0.35  deduped
│ │ └─┬ postcss-browser-comments@3.0.0
│ │   └── postcss@7.0.35  deduped
│ ├─┬ postcss-preset-env@6.7.0
│ │ ├─┬ autoprefixer@9.8.6
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ css-blank-pseudo@0.1.4
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ css-has-pseudo@0.10.0
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ css-prefers-color-scheme@3.1.1
│ │ │ └── postcss@7.0.35  deduped
│ │ ├── postcss@7.0.35  deduped
│ │ ├─┬ postcss-attribute-case-insensitive@4.0.2
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-color-functional-notation@2.0.1
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-color-gray@5.0.0
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-color-hex-alpha@5.0.3
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-color-mod-function@3.0.3
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-color-rebeccapurple@4.0.1
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-custom-media@7.0.8
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-custom-properties@8.0.11
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-custom-selectors@5.1.2
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-dir-pseudo-class@5.0.0
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-double-position-gradients@1.0.0
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-env-function@2.0.2
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-focus-visible@4.0.0
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-focus-within@3.0.0
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-font-variant@4.0.1
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-gap-properties@2.0.0
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-image-set-function@3.0.1
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-initial@3.0.4
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-lab-function@2.0.1
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-logical@3.0.0
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-media-minmax@4.0.0
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-nesting@7.0.1
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-overflow-shorthand@2.0.0
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-page-break@2.0.0
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-place@4.0.1
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-pseudo-class-any-link@6.0.0
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-replace-overflow-wrap@3.0.0
│ │ │ └── postcss@7.0.35  deduped
│ │ ├─┬ postcss-selector-matches@4.0.0
│ │ │ └── postcss@7.0.35  deduped
│ │ └─┬ postcss-selector-not@4.0.1
│ │   └── postcss@7.0.35  deduped
│ ├─┬ postcss-safe-parser@5.0.2
│ │ └── postcss@8.2.14
│ └─┬ resolve-url-loader@3.1.3
│   └── postcss@7.0.21
└─┬ sanitize-html@2.3.3
  └── postcss@8.2.9
```
